filters/openpolicyagent/evaluation.go

```diff
@@ -65,6 +69,46 @@ func (opa *OpenPolicyAgentInstance) Eval(ctx context.Context, req *ext_authz_v3.
 	return result, nil
 }
 
+func setDecisionIdInRequest(req *ext_authz_v3.CheckRequest, decisionId string) {
+	if metaDataContextDoesNotExist(req) {
+		req.Attributes.MetadataContext = formFilterMetadata(decisionId)
+	} else {
+		req.Attributes.MetadataContext.FilterMetadata["open_policy_agent"] = formOpenPolicyAgentMetaDataObject(decisionId)
+	}
+}
+
+func metaDataContextDoesNotExist(req *ext_authz_v3.CheckRequest) bool {
+	return req.Attributes.MetadataContext == nil
+}
+
+func formFilterMetadata(decisionId string) *ext_authz_v3_core.Metadata {
+	metaData := &ext_authz_v3_core.Metadata{
+		FilterMetadata: map[string]*pbstruct.Struct{
+			"open_policy_agent": {
+				Fields: map[string]*pbstruct.Value{
+					"decision_id": {
+						Kind: &pbstruct.Value_StringValue{StringValue: decisionId},
+					},
+				},
+			},
+		},
+	}
+	return metaData
+}
+
+func formOpenPolicyAgentMetaDataObject(decisionId string) *pbstruct.Struct {
+	nestedStruct := &pbstruct.Struct{}
+	nestedStruct.Fields = make(map[string]*pbstruct.Value)
+
+	innerFields := make(map[string]interface{})
+	innerFields["decision_id"] = decisionId
+
+	innerBytes, _ := json.Marshal(innerFields)
```

Comment:
This looks slightly off. You should be able to construct the struct directly without any json marshalling / unmarshalling

Reply:
See for example https://pkg.go.dev/google.golang.org/protobuf/types/known/structpb#NewStruct